### PR TITLE
Brep improvements

### PR DESCRIPTION
**Pull request by chenkasirer**
* Matched the `compas.geometry.Brep` interface with `OCCBrep`.
* Added `Brep.contains` 
* Added `BrepFace.adjacent_faces()` and `BrepFace.as_brep()` and implemented both in `RhinoBrep` 

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [x] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [x] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/geometry/brep/__init__.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+from compas.plugins import pluggable
+from compas.plugins import PluginNotInstalledError
+
+
+@pluggable(category="factories")
+def new_brep(cls, *args, **kwargs):
+    raise PluginNotInstalledError
+
+
+@pluggable(category="factories")
```
Comment (tomvanmele) [inline src/compas/geometry/brep/__init__.py:10]: not a must, but would be cool if these could be alphabetical

File: src/compas/geometry/brep/edge.py [hunk 2]
```diff
@@ -22,16 +40,16 @@ class BrepEdge(Data):
         Returns True if this edge is a bspline, False otherwise.
     is_other : bool, read-only
         Returns True if this edge is of another shape, False otherwise.
-    is_valid : bool, read-only
-        Returns True if this edge is valid, False otherwise.
+    orientation : literal(BrepOrientation), read-only
```
Comment (tomvanmele) [inline src/compas/geometry/brep/edge.py]: :class:`BrepOrientation`

File: src/compas/geometry/brep/edge.py [hunk 3]
```diff
@@ -22,16 +40,16 @@ class BrepEdge(Data):
...
+        Returns the orientation of this edge. One of: FORWARD, REVERSED, INTERNAL, EXTERNAL.
+    type : literal(CurveType), read-only
```
Comment (tomvanmele) [inline src/compas/geometry/brep/edge.py]: :class:`CurveType`

File: src/compas_rhino/geometry/brep/face.py [hunk 4]
```diff
@@ -126,7 +145,53 @@ def holes(self):
...
+        elif self.is_plane:
+            return SurfaceType.PLANE
+        else:
+            return SurfaceType.OTHER_SURFACE
+
+    # ==============================================================================
+    # Methods
+    # ==============================================================================
 
     @staticmethod
```
Comment (tomvanmele) [inline src/compas_rhino/geometry/brep/face.py:198]: why is this a static method?

File: src/compas_rhino/geometry/brep/face.py [hunk 5]
```diff
@@ -50,6 +54,7 @@ def __init__(self, rhino_face=None, builder=None):
 
     def _set_face(self, native_face):
```
Comment (tomvanmele) [inline src/compas_rhino/geometry/brep/face.py]: why not make this a property with a setter?

File: src/compas/geometry/brep/__init__.py (hunk 1 above)
Comment (chenkasirer) [reply src/compas/geometry/brep/__init__.py:10]: the hardest part of this entire re-work, I find ;)

File: src/compas_rhino/geometry/brep/face.py (hunk 4 above)
Comment (chenkasirer) [reply src/compas_rhino/geometry/brep/face.py:198]: it's oblivious to the instance. it just does some conversions on the given surface.

File: src/compas_rhino/geometry/brep/face.py (hunk 5 above)
Comment (chenkasirer) [reply src/compas_rhino/geometry/brep/face.py]: cool idea. re-worked this system [here](https://github.com/compas-dev/compas/pull/1224/commits/2607834a0d2ae90d89ec23ff0174882a33dbb272).

**Review by tomvanmele — APPROVED** [review]

**Pull request merged**